petals-se-camel/src/main/java/org/ow2/petals/camel/se/CamelSU.java

```diff
@@ -176,9 +178,24 @@ public void undeploy() {
         }
     }
 
+    /**
+     * Notifies placeholders reloading, only to Camel routes based on {@link PetalsRouteBuilder}
+     * 
+     * @param placeholders
+     *            New values of placeholders
+     */
+    public void onPlaceHolderValuesReloaded(final Properties placeholders) {
+        for (final RouteBuilder routeBuilder : this.classRoutes) {
+            assert routeBuilder != null;
+            if (routeBuilder instanceof PetalsRouteBuilder) {
```

Comment:
Either all method calls should go through an `instanceof` (and thus you remove the `callMethod` method) or all method calls should go through `callMethod` (so that the user is not forced to use `PetalsRouteBuilder`).
But not a mix of the two :)

Reply:
It was more simpler to do as than creating new methods to use Java reflection ;)

Reply:
yep but you do have to choose between one or the other: you can't mix the two, it's bad design!
So either do this for all method calls and remove `callMethods` or remove this and improve `callMethods`.